### DISCUSSION TITLE
Packed bool arrays aka bitsets/bitvectors support

### DISCUSSION THREAD

**Discussion by Nemoumbra** [NationalSecurityAgency/ghidra#6981] [Q&A]
I'm reversing a program that allocates a memory range where it stores its state flags.
Example use scenario:
```c++
pPVar2 = get_flags_ptr(&pGVar1->substruct);
pPVar2->flags[0x1cd] = pPVar2->flags[0x1cd] & 0xef;
```

I've assigned a custom type `ProgramStateFlags` to the variable `pPVar2`. This struct has fields for the size and the `byte*` pointer to the allocated memory zone `flags`. As you could have guessed, the flags are stored in the compressed form that optimizes the size.

----

I can now manually figure out what's going on here by
- converting the constant `0xef` to `0b11101111`, checking that it has a single bit set to false, that is, the bit with index 4 (right to left)
- evaluating `0x1cd * 8 + 4 == 0xE6C == 3692`
- adding a pre comment saying `disable FLAG[3692];`

Is this the best option I've got? I was hoping for some bitvector/bitset types.

**Comment by Wall-AF** [comment]
Use an `enum`.  E.g.:
```
enum xxxFlags {
   flg1=1,
   flg2=2,
   flg3=4,
   ...
   flg8=128
}
```
